numpy/_core/src/umath/loops_umath_fp.dispatch.c.src

```diff
@@ -250,6 +251,25 @@ NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(@TYPE@_@func@)
         simd_@intrin@_@sfx@(src1, ssrc1, src2, ssrc2, dst, sdst, len);
         return;
     }
+#endif
+#if @ispower@
+    if (steps[1]==0) {
+        BINARY_DEFS_ZERO_STRIDE
```

Comment:
Would it make sense to call the inner loop for `square`? Mostly if that helps with the CPU dispatch (I don't understand that well enough to know whether this is possible).

Reply:
We can update this to use SIMD for square and sqrt in another patch.